tdrs-backend/tdpservice/search_indexes/parsers/tanf_parser.py

```diff
@@ -0,0 +1,97 @@
+"""Transforms a TANF datafile into an search_index model."""
+
+import logging
+from ..models import T1  # , T2, T3, T4, T5, T6, T7, ParserLog
+# from django.core.exceptions import ValidationError
+from .util import get_record_type
+from .schema_defs.tanf import t1_schema
+
+logger = logging.getLogger(__name__)
+logger.setLevel(logging.DEBUG)
+
+def active_t1_parser(line, line_number):
+    """Parse line in datafile as active case data, T1 only."""
+    family_case_schema = t1_schema()
+    # create search_index model
+    t1 = T1()
+    content_is_valid = True
+
+    expected_line_length = 156  # we will need to adjust for other types
+    actual_line_length = len(line)
+    if actual_line_length != expected_line_length:
+        logger.error('Expected line length of 156, got: %s', actual_line_length)
```

Comment:
@andrew-jameson I'm noticing that this error is raised when there are trailing spaces in T1. Will this prevent parsing? if so, can we account for this?  There are 2 scenarios where T1 records have trailing spaces:

1. disposition == 2 (the case was reported in error). in this case, the STT doesnt have to report on the rest of the data elements, When this happens, there will be 125 trailing spaces (or zeros) to get len==156.

> 9. Disposition:
Guidance: A family that did not receive any assistance for the reporting month but was listed on
the monthly sample frame for the reporting month is "listed in error." States must collect and
report complete data for all sampled cases that are not listed in error. If the family is "listed in
error, code this data element with a "2" and do not complete data elements # 10 through # 77.
Instruction: Enter one of the following codes for each TANF sampled case.
1 = Data collection completed
2 = Not subject to data collection/listed in error

2. While some STTs pad the end of their T1 records with 39 zeros, many will typically pad with trailing spaces to ensure that the len==156, per the [T1 record layout](https://www.acf.hhs.gov/sites/default/files/documents/ofa/tanf_data_report_section1_10_2008.pdf).

Reply:
The code currently discards whats in the range 118-156 AFTER the length check; that is to say, whether it's zeroes or spaces I don't think matters. Given the disposition case, we'll have to refactor the code and add a new unit test to cover it. I'm imagining we can make this case a "don't care" condition where if disposition==2 then we won't enforce length at all.

Reply:
@andrew-jameson

- re: i like your suggestion re: disposition -- agree, we dont need to enforce the len condition
- re: 118 - 156 --makes sense 
- what about when T1 records are too long? currently the following record was not parsed, but id expect that itll parse anything up to 156 and discard everything that comes after it. not required, but just wanted to be sure what we implemented for this scenario. 

```
2023-01-20T12:19:48.09-0500 [APP/PROC/WEB/0] ERR [2023-01-20 17:19:48,090 DEBUG util.py::get_record_type:L17 :  Matched following line as data: T12025101111111114721801401711120212110374300000000000003820060000000000000000000000000000000000222222000000002229012
2023-01-20T12:19:48.09-0500 [APP/PROC/WEB/0] ERR [2023-01-20 17:19:48,090: DEBUG/ForkPoolWorker-67] Matched following line as data: T12025101111111114721801401711120212110374300000000000003820060000000000000000000000000000000000222222000000002229012
2023-01-20T12:19:48.09-0500 [APP/PROC/WEB/0] ERR [2023-01-20 17:19:48,090 ERROR tanf_parser.py::active_t1_parser:L22 :  Expected line length of 156, got: 158
2023-01-20T12:19:48.09-0500 [APP/PROC/WEB/0] ERR [2023-01-20 17:19:48,090: ERROR/ForkPoolWorker-67] Expected line length of 156, got: 158

```

Reply:
Looking over both the length and disposition cases, I have implemented a similar "don't care" for padded length as with a trailer (T1 >= 118).